lib/rdoc/rubygems_hook.rb

```diff
@@ -281,14 +281,17 @@ def initialize(spec, generate_rdoc = false, generate_ri = true)
       @spec = spec
       @generate_rdoc = generate_rdoc
       @generate_ri   = generate_ri
```

Comment:
I was thinking that we can initialize a `RubyGemsHook` here already, like

```rb
    def initialize(spec, generate_rdoc = false, generate_ri = true)
      @rubygems_hook = RubyGemsHook.new(spec, generate_rdoc, generate_ri)
    end
```

And then delegate methods to `@rubygems_hook`, which will handle more cases 🤔

Reply:
IMO, implementing minimal things to clarify what method/attribute are needed in the compatibility layer is not a bad idea. I think it is the intention of the current implementation of RubygemsHook. (I', not sure)
This way, the new RubyGemsHook can easily change its method as long as RubygemsHook's methods works.